src/GitTools.Core/Context/AuthenticationContext.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace GitTools
+{
+    public class AuthenticationContext : IAuthenticationContext
+    {
+        public AuthenticationContext()
+        {
+        }
+
+        public AuthenticationContext(string username, string password)
+        {
+            Username = username;
+            Password = password;
+        }
+
+        public AuthenticationContext(string token)
+        {
+            Token = token;
+        }
+
+        public string Username { get; set; }
+        public string Password { get; set; }
```

Comment:
In all honesty, I don't know :smile_cat: It just occurred to me that perhaps we should.  If we can't accept the input from the command line as a SecureString, then there isn't much point (I think) in converting as the string would already to subject to "attack".  Not sure there is much to be gained here, unless we prompt the user to input credentials into standard credential window, rather than directly at the command line, but that would break the automation route?!?